client/ayon_core/pipeline/staging_dir.py

```diff
@@ -11,21 +14,41 @@
 @dataclass
 class StagingDir:
     directory: str
-    persistent: bool
-    custom: bool  # Whether the staging dir is a custom staging dir
+    is_persistent: bool
+    # Whether the staging dir is a custom staging dir
+    is_custom: bool
+
+    def __setattr__(self, key, value):
+        if key == "persistent":
+            warnings.warn(
+                "'StagingDir.persistent' is deprecated."
+                " Use 'StagingDir.is_persistent' instead.",
+                DeprecationWarning
+            )
+            key = "is_persistent"
+        super().__setattr__(key, value)
+
+    @property
+    def persistent(self):
+        warnings.warn(
+            "'StagingDir.persistent' is deprecated."
+            " Use 'StagingDir.is_persistent' instead.",
+            DeprecationWarning
+        )
+        return self.is_persistent
 
 
 def get_staging_dir_config(
-    project_name,
-    task_type,
-    task_name,
-    product_type,
-    product_name,
-    host_name,
-    project_settings=None,
-    anatomy=None,
-    log=None,
-):
+    project_name: str,
+    task_type: Optional[str, None],
+    task_name: Optional[str, None],
```

Comment:
```suggestion
    task_type: Optional[str],
    task_name: Optional[str],
```

I got the following error without this change.
```
    from ayon_core.pipeline import install_ayon_plugins
  File "C:\Users\robin\OneDrive\Bureau\dev_ayon\dev\ayon-core\client\ayon_core\pipeline\__init__.py", line 12, in <module>
    from .staging_dir import get_staging_dir_info
  File "C:\Users\robin\OneDrive\Bureau\dev_ayon\dev\ayon-core\client\ayon_core\pipeline\staging_dir.py", line 43, in <module>
    task_type: Optional[str, None],
  File "C:\Python\Python39\lib\typing.py", line 277, in inner
    return func(*args, **kwds)
  File "C:\Python\Python39\lib\typing.py", line 354, in __getitem__
    return self._getitem(self, parameters)
  File "C:\Python\Python39\lib\typing.py", line 478, in Optional
    arg = _type_check(parameters, f"{self} requires a single type.")
  File "C:\Python\Python39\lib\typing.py", line 166, in _type_check
    raise TypeError(f"{msg} Got {arg!r:.100}.")
TypeError: typing.Optional requires a single type. Got (<class 'str'>, None).
```
https://stackoverflow.com/questions/73893814/optional-typing-for-more-than-a-single-type